### ISSUE TITLE
Multiple Channel are created if move from one n/w (Wifi) to flight mode and then move to another n/w. it is clearing only after 2 min mean while some messages delivered to the previous chanel is blocked it that channel is cleared.

### ISSUE THREAD

**Issue by balaramachandrant** [rabbitmq/rabbitmq-objc-client#118]
Below are my observations and need help in resolving it.
I declared message queue with out any queue  options.

 MSG_conn = [[RMQConnection alloc] initWithUri:uri tlsOptions:[RMQTLSOptions fromURI:uri] channelMax:@(RMQChannelLimit) frameMax:@(RMQFrameMax) heartbeat:@0 syncTimeout:@4 delegate:[RMQConnectionDelegateLogger new] delegateQueue:dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)];

		[MSG_conn start];
		//Create Channel
		ch = [MSG_conn createChannel];
		MSG_queue=[ch queue:Msg_queuename];

Initially I was in Wifi exchanged couple of messages say 1 to 5. Later I moved to flight mode in that mode I received 5 more messages say 6 to 10 .  rmq server delivered the messages to the consumer. when I entered into another n/w new consumer I started receiving messages from 11 to 15 . I got the message from 6 to 10 only after 2 min when the old consumer got destroyed. How to reduce the delay time

**Comment by michaelklishin** [comment]
Thank you for your time.

Team RabbitMQ uses GitHub issues for specific actionable items engineers can work on. This assumes we have a certain amount of information to work with. Questions, investigations, root cause analysis, discussions for potential features are all considered to be mailing list material by our team. When/if we have enough details and evidence we'd be happy to file a new issue.

Please post this to [rabbitmq-users](https://groups.google.com/forum/#!forum/rabbitmq-users). Thank you.

**Comment by michaelklishin** [comment]
Welcome to distributed systems. [Connection unavailability is not detected instantly](http://www.rabbitmq.com/heartbeats.html) in many cases.
